client/src/Component/Main/AddVideo.jsx

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+
+const AddVideo = ({ addVideo }) => {
+  const [error, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState({});
+  const [formInput, setFormInput] = useState({
+    title: "",
+    url: "",
+  });
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    const missingObject = Object.entries(formInput)
+      .filter(([key, value]) => {
+        return !value;
+      })
+      .map(([key, value]) => {
+        return ` the ${key} is missing`;
+      });
+
+    if (missingObject.length >= 1) {
+      setError(true);
+      setErrorMessage(missingObject);
+      return;
+    }
+
+    if (!isValidYouTubeUrl(formInput.url)) {
+      setError(true);
+      setErrorMessage(["please insert a valid youtube url"]);
+      return;
+    }
+
+    setError(false);
+    addVideo(formInput);
+    setFormInput({
+      title: "",
+      url: "",
+    });
+  };
+
+  function isValidYouTubeUrl(url) {
+    let youtubePattern =
+      /^(https?:\/\/)?(www\.)?youtube\.com\/watch\?v=[\w-]{10}[A-Z]$/;
+    return youtubePattern.test(url);
+  }
```

Comment:
Nice, breaking it into modular functions that could be reused. It would be best to create this function outside of a component then you could use it elsewhere 😄

Reply:
I added to main component and I use it as props , but I am not sure if it is correct solution